server/user/user.controller.js

```diff
@@ -44,9 +44,32 @@ function login({ body }, res) {
     });
 }
 
+function updateProfile({ user, body: { userCredentials } }, res, next) {
+  const { email, firstName, lastName } = userCredentials;
+  return user.update({ email, firstName, lastName })
+    .then(({ profile }) => res.json({ user: profile }))
+    .catch(err => next(err));
+}
+
+function changePassword({ user, body: { currentPassword, newPassword } }, res) {
+  return user.authenticate(currentPassword)
+    .then(user => user || createError(NOT_FOUND, 'Incorrect current password'))
+    .then(user => user.update({ password: newPassword }))
+    .then(() => res.sendStatus(200));
+}
+
+function updateImageUrl({ user, body: { key } }, res, next) {
```

Comment:
And where did you get an url from? 😕 This is not upload, this is pasting public image link?